### ISSUE TITLE
Uno.WinUI TextBox IsEnabled inheritance under Skia.Gtk

### ISSUE THREAD

**Issue by brinko99** [unoplatform/uno#13887]
### Current behavior

It is expected that controls inherit the `IsEnabled` false state from their context. Uno's Skia TextBox implementation allows an `IsEnabled` false state to be overridden as shown:

```
<ContentControl IsEnabled="False">
    <TextBox IsEnabled="True" Text="Should be disabled" />
</ContentControl>
```

Under Skia.Gtk (Linux), this produces a `TextBox` that is enabled. It should be disabled as-is under Windows Desktop [and Wasm](http://playground.platform.uno/#847e2503).

Further, in the following situation, the `TextBox` is effectively disabled (cannot edit), but does not display the proper styling of a disabled `TextBox`:

```
<ContentControl IsEnabled="False">
    <ContentControl IsEnabled="True">
        <TextBox Text="Should be disabled" />
    </ContentControl>
</ContentControl>
```
![image](https://github.com/unoplatform/uno/assets/1231765/ba70d81a-58e9-4458-927c-74f8d4a65adf)
(note, thicker bottom border)
(apologies for the image quality)

Actual disabled `TextBox` for reference:
![image](https://github.com/unoplatform/uno/assets/1231765/16d1390c-db06-4cec-b238-85e9b2a2c3f6)

In our real app, a `UserControl` is being disabled by a parent, yet individual controls within the `UserControl` are enabled due to binding (e.g. "is valid"). This is a pretty big deal for us.

Thanks

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

version 4.10.13

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

**Comment by Youssef1313** [comment]
Thanks for filing this @brinko99.
I think this issue was already fixed by #12884, though we will not be able to backport it to 4.x. Can you please upgrade to the latest 5.0 dev packages? Let us know if you have any issues with the upgrade.

**Comment by brinko99** [comment]
Thanks for the update...  We're not in a position to update to 5.0 dev at this time.

The fix does look promising. We'll test with it as soon as possible; though likely not until 5.0 ship. We'll reopen the issue if the problem persists at that time.
